router/transformer/transformer.go

```diff
@@ -42,9 +48,9 @@ type HandleT struct {
 
 // Transformer provides methods to transform events
 type Transformer interface {
-	Setup()
+	Setup(timeout time.Duration)
 	Transform(transformType string, transformMessage *types.TransformMessageT) []types.DestinationJobT
-	ProxyRequest(ctx context.Context, responseData integrations.PostParametersT, destName string, jobId int64) (statusCode int, respBody string)
+	ProxyRequest(ctx context.Context, responseData integrations.PostParametersT, destName string, jobId int64, varargs ...interface{}) (statusCode int, respBody string)
```

Comment:
The variable arguments changes made to support testing sounded vague to me. Can you find alternatives?

Reply:
Looks good, but unfortunately, it adds a `marshal` operation. How is this problem solved in other unit tests? Is this unique to this situation?
`payload, err := jsonfast.Marshal(proxyReqParams.ResponseData)`

Reply:
The marshalling was already present even before this refactor

Reply:
My bad.